src/iconv.jl

```diff
@@ -102,11 +108,11 @@ function iconv_reset!(s::Union{StringEncoder, StringDecoder})
                 (Ptr{Void}, Ptr{Ptr{UInt8}}, Ref{Csize_t}, Ptr{Ptr{UInt8}}, Ref{Csize_t}),
                 s.cd, C_NULL, C_NULL, s.outbufptr, s.outbytesleft)
 
-    if ret == reinterpret(Csize_t, -1)
+    if ret == typemax(Csize_t)
```

Comment:
I changed that because I don't think reinterpret(Csize_t, -1) will work on a 32-bit machine.
reinterpret(Csize_t, Int32(-1)) fails, and Int is Int32 on 32-bit machines.